x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable_factory.tsx

```diff
@@ -0,0 +1,350 @@
+              if (selectedJobId !== undefined && selectedJob !== undefined) {
+                zoomDuration = mlTimeSeriesExplorerService?.getAutoZoomDuration(selectedJob);
+                setAutoZoomDuration(zoomDuration);
+              }
+            },
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+            [selectedJobId, selectedJob?.job_id, mlTimeSeriesExplorerService]
+          );
+
+          useEffect(function onUnmount() {
+            return () => {
+              subscriptions.unsubscribe();
+            };
+          }, []);
+
+          // eslint-disable-next-line react-hooks/exhaustive-deps
+          const resizeHandler = useCallback(
+            throttle((e: { width: number; height: number }) => {
+              // Keep previous container height so it doesn't change the page layout
+              containerHeightRef.current = e.height;
+
+              if (Math.abs(chartWidth - e.width) > 20) {
+                setChartWidth(e.width);
+              }
+            }, RESIZE_THROTTLE_TIME_MS),
+            [chartWidth]
+          );
+
+          const appStateHandler = useCallback(
+            (action: string, payload?: any) => {
+              /**
+               * Empty zoom indicates that chart hasn't been rendered yet,
+               * hence any updates prior that should replace the URL state.
+               */
+
+              switch (action) {
+                case APP_STATE_ACTION.SET_FORECAST_ID:
+                  setSelectedForecastId(payload);
+                  setZoom(undefined);
+                  break;
+
+                case APP_STATE_ACTION.SET_ZOOM:
+                  setZoom(payload);
+                  break;
+
+                case APP_STATE_ACTION.UNSET_ZOOM:
+                  setZoom(undefined);
+                  break;
+              }
+            },
+
+            [setZoom, setSelectedForecastId]
+          );
+
+          if (error) {
+            return (
+              <EuiCallOut
+                title={
+                  <FormattedMessage
+                    id="xpack.ml.singleMetricViewerEmbeddable.errorMessage"
+                    defaultMessage="Unable to load the ML single metric viewer data"
+                  />
+                }
+                color="danger"
+                iconType="warning"
+                css={{ width: '100%' }}
+              >
+                <p>{error}</p>
+              </EuiCallOut>
+            );
+          }
+
+          return (
+            <I18nContext>
+              <KibanaThemeProvider theme={theme}>
+                <KibanaContextProvider
+                  services={{
+                    mlServices: {
+                      ...services[2],
+                    },
+                    ...services[0],
+                    ...services[1],
+                  }}
+                >
+                  <DatePickerContextProvider {...datePickerDeps}>
+                    <Suspense fallback={<EmbeddableLoading />}>
```

Comment:
we don't need a Suspense here 
```suggestion
```

Reply:
Removed in 69379adfcb65a3f1971a186c5be07988f41e721e